metrics_utility/automation_controller_billing/collectors.py

```diff
@@ -551,10 +552,17 @@ def total_workers_vcpu(since, full_path, until, **kwargs):
         raise MissingRequiredEnvVar('environment variable METRICS_UTILITY_CLUSTER_NAME is not set')
 
     now = datetime.now(timezone.utc)
-
-    info = {'cluster_name': cluster_name, 'timestamp': now.isoformat(), 'nodes': []}
-    # If METRICS_UTILITY_USAGE_BASED_BILLING_ENABLED is not set or set to false then it returns 1
-    usage_based_billing_enabled_str = os.getenv('METRICS_UTILITY_USAGE_BASED_BILLING_ENABLED')
+    current_ts = now.timestamp()
+    prev_hour_start, prev_hour_end, current_hour_start = get_hour_boundaries(current_ts)
+
+    info = {
+        'cluster_name': cluster_name,
+        'timestamp': datetime.fromtimestamp(prev_hour_end).isoformat(),
+        'start_timestamp': datetime.fromtimestamp(prev_hour_start).isoformat(),
+        'end_timestamp': datetime.fromtimestamp(current_hour_start).isoformat(),
```

Comment:
It seems to me the three timestamps we care about are:
1. When did we take the collection? current_ts
2. When did the collection window start? prev_hour_start
3. When did the collection window end? prev_hour_end

As is, listing the `current_hour_start` as the  `end_timestamp` gives the impression that we have overlapping collection windows that could lead to double-counting in extreme edge cases (not as currently coded, but gives the appearance as such, and collection windows should not overlap).
What do you think?

Reply:
ok, I found it was more understable for PM to have entire hours rather than x:59:59 but understand your point. I will change the code to show from x:00:00 to x:59:59